flex_config/file_sources/file_source.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+from typing import Any, Dict, Iterable, TextIO, Tuple
+
+from flex_config.config_source import ConfigSource
+
+
+class FileSource(ConfigSource):
+    """
+    A ConfigSource that loads values from a file. Should not be used directly - use one its subclasses for the supported
+        file types.
+
+    Args:
+        path (pathlib.Path): a [pathlib.Path](https://docs.python.org/3/library/pathlib.html#basic-use) to a file
+            (of a supported type) to load.
+        path_must_exist (bool): if true, the constructor will raise a RuntimeError if the provided path doesn't exist.
```

Comment:
Document the exception this can raise under the `Raises:` header. Also I would make it a `FileExistsError` to be more semantic.

```suggestion
        path_must_exist (bool): Enforce that the passed path must exist. If not, default to empty source.
        
    Raises:
        FileExistsError: If path_must_exist is true but the path does not exist
```